Gordon360/Services/RecIM/SeriesService.cs

```diff
@@ -0,0 +1,583 @@
+            };
+            await _context.SeriesSchedule.AddAsync(schedule);
+            await _context.SaveChangesAsync();
+
+            if (seriesSchedule.SeriesID is not null)
+            {
+                var series = _context.Series.FirstOrDefault(s => s.ID == seriesSchedule.SeriesID);
+                series.ScheduleID = schedule.ID;
+            }
+            await _context.SaveChangesAsync();
+            return schedule;
+
+        }
+        public async Task<SeriesViewModel> UpdateSeriesAsync(int seriesID, SeriesPatchViewModel update)
+        {
+            var s = await _context.Series.FindAsync(seriesID);
+            s.Name = update.Name ?? s.Name;
+            s.StartDate = update.StartDate ?? s.StartDate;
+            s.EndDate = update.EndDate ?? s.EndDate;
+            s.StatusID = update.StatusID ?? s.StatusID;
+
+            await _context.SaveChangesAsync();
+            return s;
+        }
+        public async Task UpdateSeriesTeamStats(SeriesTeamPatchViewModel update)
+        {
+            var st = await _context.SeriesTeam.FindAsync(update.ID);
+            st.Win = update.Win ?? st.Win;
+            st.Loss = update.Loss ?? st.Loss;
+
+            await _context.SaveChangesAsync();
+        }
+
+        private async Task CreateSeriesTeamMappingAsync(IEnumerable<int> teams, int seriesID)
+        {
+            foreach (var teamID in teams)
+            {
+                var seriesTeam = new SeriesTeam
+                {
+                    TeamID = teamID,
+                    SeriesID = seriesID,
+                    Win = 0,
+                    Loss = 0
+                };
+                await _context.SeriesTeam.AddAsync(seriesTeam);
+            }
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task DeleteSeriesCascadeAsync(int seriesID)
+        {
+            //delete series teams
+            var seriesTeam = _context.SeriesTeam.Where(st => st.SeriesID == seriesID);
+            _context.SeriesTeam.RemoveRange(seriesTeam);
+            //delete series surfaces
+            var seriesSurface = _context.SeriesSurface.Where(ss => ss.SeriesID == seriesID);
+            _context.SeriesSurface.RemoveRange(seriesSurface);
+            //delete matches
+            var matchIDs = _context.Match.Where(m => m.SeriesID == seriesID).Select(m => m.ID).ToList();
+            foreach (var matchID in matchIDs)
+            {
+                await _matchService.DeleteMatchCascadeAsync(matchID);
+            }
+            //delete series
+            var series = _context.Series.FirstOrDefault(s => s.ID == seriesID);
+            _context.Series.Remove(series);
+            await _context.SaveChangesAsync();
+        }
+
+        // Scheduler does not currently handle overlaps
+        // eventually:
+        // - ensure that matches that occur within 1 hour do not share the same surface
+        //    unless they're in the same series
```

Comment:
This comment should be an XML Doc Comment.

Reply:
This was a personal note for myself, but I could make it more professional in the event am unable to return to it.